src/dbnode/storage/mediator.go

```diff
@@ -135,39 +140,22 @@ func (m *mediator) EnableFileOps() {
 	m.databaseFileSystemManager.Enable()
 }
 
-// Tick mediates the relationship between ticks and flushes/snapshots/cleanups.
+// The mediator mediates the relationship between ticks and flushes(warm and cold)/snapshots/cleanups.
 //
 // For example, the requirements to perform a flush are:
 // 		1) currentTime > blockStart.Add(blockSize).Add(bufferPast)
 // 		2) node is not bootstrapping (technically shard is not bootstrapping)
-// 		3) at least one complete tick has occurred since blockStart.Add(blockSize).Add(bufferPast)
-//		4) at least one complete tick has occurred since bootstrap completed (can be the same tick
-// 		   that satisfies condition #3)
-//
-// The mediator helps ensure conditions #3 and #4 are met by measuring the tickStart time and the shard bootstrap
-// states *before* the tick, and then propagating those values to downstream components so they can use that
-// information to make decisions about whether to flush / snapshot / run cleanups.
 //
-// Measuring the tickStart before the tick and propagating that also helps the tick and flush logic to coordinate.
 // For example, there is logic in the Tick flow for removing shard flush states from a map so that it doesn't
 // grow infinitely for nodes that are not restarted. If the Tick path measured the current time when it made that
 // decision instead of using the same measurement that is shared with the flush logic, it might end up removing
 // a shard flush state (due to it expiring), but since the flush logic is using a slightly more stale timestamp it
 // will think that the old block hasn't been flushed (even thought it has) and try to flush it even though the data
 // is potentially still on disk (if it hasn't been cleaned up yet).
-func (m *mediator) Tick(runType runType, forceType forceType) error {
-	tickStart := m.nowFn()
-	dbBootstrapStateAtTickStart := m.database.BootstrapState()
-
-	if err := m.databaseTickManager.Tick(forceType, tickStart); err != nil {
-		return err
-	}
-
-	// NB(r): Cleanup and/or flush if required to cleanup files and/or
-	// flush blocks to disk. Note this has to run after the tick as
-	// blocks may only have just become available during a tick beginning
-	// from the tick begin marker.
-	m.databaseFileSystemManager.Run(tickStart, dbBootstrapStateAtTickStart, syncRun, forceType)
+//
+// See comment over mediatorTimeBarrier for more details on how this is implemented.
+func (m *mediator) RunFilesystemProcesses(forceType forceType, startTime time.Time) error {
+	m.databaseFileSystemManager.Run(startTime, syncRun, forceType)
```

Comment:
Just wondering, does this method make sense to have a single line method wrapper?